specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-06-02-preview/managedClusters.json

```diff
@@ -7763,6 +7774,69 @@
         "enabled"
       ]
     },
+    "IstioCertificateAuthority": {
+      "type": "object",
+      "description": "Istio Service Mesh Certificate Authority (CA) configuration. For now, we only support plugin certificates as described here https://aka.ms/asm-plugin-ca",
+      "properties": {
+        "plugin": {
+          "$ref": "#/definitions/IstioPluginCertificateAuthority"
+        }
+      }
+    },
+    "IstioPluginCertificateAuthority": {
+      "type": "object",
+      "description": "Plugin certificates information for Service Mesh.",
+      "properties": {
+        "keyVaultId": {
+          "type": "string",
+          "format": "arm-id",
+          "x-ms-arm-id-details": {
+            "allowedResources": [
+              {
+                "type": "Microsoft.KeyVault/vaults"
+              }
+            ]
+          },
+          "description": "The resource ID of the Key Vault."
+        },
+        "certObjectName": {
+          "type": "string",
+          "description": "Intermediate certificate object name in Azure Key Vault."
+        },
+        "keyObjectName": {
+          "type": "string",
+          "description": "Intermediate certificate private key object name in Azure Key Vault."
+        },
+        "rootCertObjectName": {
+          "type": "string",
+          "description": "Root certificate object name in Azure Key Vault."
+        },
+        "certChainObjectName": {
+          "type": "string",
+          "description": "Certificate chain object name in Azure Key Vault."
+        }
+      }
+    },
+    "IstioEgressGateway": {
+      "type": "object",
+      "description": "Istio egress gateway configuration.",
+      "properties": {
+        "enabled": {
+          "type": "boolean",
+          "description": "Whether to enable the egress gateway."
+        },
+        "nodeSelector": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "NodeSelector for scheduling the egress gateway."
```

Comment:
There's a [LintDiff error](https://github.com/Azure/azure-rest-api-specs/pull/24453/checks?check_run_id=14277272454) about this block. A recently added rule [AvoidAdditionalProperties](https://github.com/Azure/azure-openapi-validator/blob/main/docs/avoid-additional-properties.md) attempts to limit such usage. In addition, I checked the swagger of other services, and there are not many similar usages (except for tags).

If you insist on using such a structure (map[string]string), there is a high probability that you need to clarify the specific reasons to the ARM reviewer when performing the final ARM review before the swagger release.

Reply:
Maps are pretty standard in JSON - I'm surprised the Swagger team is pushing this rule? What's the reasoning behind it? The Azure guidelines state the exact opposite of this:

See [here](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md)

> [✔️](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#json-use-arrays-for-ordering) YOU MAY use JSON arrays if maintaining an order of values is required. Avoid arrays in other situations since arrays can be difficult and inefficient to work with, especially with JSON Merge Patch where the entire array needs to be read prior to any operation being applied to it.

The only alternative to this map structure here would be an array, right?

Reply:
@FumingZhang @matthchr this is in line with nodeLabels under AgentPoolProperties: https://github.com/Azure/azure-rest-api-specs/blob/e2b5f9323c4214408969a6e953b4075cfdc693b6/specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2023-03-01/managedClusters.json#L2828

Reply:
`Definitions must not have properties named "additionalProperties" except for user defined tags.`

^ I think labels used for NodeSelectors would fall under here?